packages/ui-components/src/ExpandableText/ExpandableText.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { useEffect, useState } from "react";
+import styles from "./ExpandableText.module.scss";
+import classNames from "classnames/bind";
+import LinesEllipsis from "react-lines-ellipsis";
+
+interface ExpandableTextProps {
+  text: string;
+  maxLine: number;
+}
+
+const cx = classNames.bind(styles);
+
+export function ExpandableText({
+  text,
+  maxLine,
+}: ExpandableTextProps): React.ReactElement {
+  const [isExpanded, setIsExpanded] = useState(false);
+
+  useEffect(() => {
+    // When the text changes, reset to the default of isExpanded=false.
+    // This is needed in case the text changes without re-mounting the component.
+    setIsExpanded(false);
+  }, [text]);
+
+  return (
+    <>
+      {!isExpanded && (
+        <LinesEllipsis
+          text={text}
+          maxLine={maxLine}
+          ellipsis={
```

Comment:
Thanks everyone!

@nathanstilwell, yeah the [type definition](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react-lines-ellipsis/index.d.ts#L14) for the `ellipsis` prop is not up-to-date with the docs, and this is generating a typescript error. What do you mean by "disabling eslint anyway"? It looks like the CI runs `tsc`, which is failing (and is a helpful guard to have always passing?)

Reply:
Sorry, I was thinking about this wrong earlier (I mixed up `eslint-ignore-next-line` with `@ts-ignore`). Ultimately while it may not be better than `as any` (and disabling the linter), you could just cast the `ellipsis` block as a `string`.

```jsx
ellipsis={
  (
    <>
      {"… "}
      <a
        className={cx("crl-anchor")}
        onClick={() => {
          setIsExpanded(true);
        }}
      >
        Show More
      </a>
    </>
  ) as unknown as string
}

```